lark/tree.pyi

```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
+
+from typing import List, Callable, Iterator, Union, Optional
+from .lexer import Token
+
+
+class Tree:
+
+    data: str
+    children: List[Union[str, Tree]]
+    meta: Token
+
+    def __init__(self, data: str, children: List[Tree], meta: Optional[Token] = None):
```

Comment:
`children` should accept the same type as `Tree.children`

`meta` is not a `Token`, it's `tree.Meta`.

Reply:
Sorry, because I saw [doc](https://lark-parser.readthedocs.io/en/latest/classes/#tree) mentioned it has some attributes not listed.
Should I add in `tree.Meta` (as dynamic attributes) or implement as a [`typing.Protocol`](https://docs.python.org/3/library/typing.html#typing.Protocol)?

Reply:
I'm not sure how Protocol will help. I think it's fine to write `Meta`'s attributes as if they are there, since they usually will be (unless empty is True)